python/lolopy/learners.py

```diff
@@ -351,38 +354,142 @@ def __init__(self, num_trees=-1, use_jackknife=True, bias_learner=None,
+    """Random Forest model used for classification"""
 
+    def __init__(self, num_trees=-1, use_jackknife=False, leaf_learner=None, subset_strategy="auto",
+                 min_leaf_instances=1, max_depth=2**30, randomize_pivot_location=False, randomly_rotate_features=False):
+        """Initialize the RandomForest
 
-class RandomForestClassifier(BaseLoloClassifier, RandomForestMixin):
-    """Random Forest model used for classification"""
+        Args:
+            num_trees (int): Number of trees to use in the forest (default of -1 sets the number of trees to the number of training rows)
+            use_jackknife (bool): Whether to use jackknife based variance estimates
+            leaf_learner (BaseLoloLearner): Learner used at each leaf of the random forest (default: GuessTheMean)
+            subset_strategy (Union[string,int,float]): Strategy used to determine number of features used at each split
+                Available options:
+                    "auto": Use the default for lolo (all features for regression, sqrt for classification)
+                    "log2": Use the base 2 log of the number of features
+                    "sqrt": Use the square root of the number of features
+                    integer: Set the number of features explicitly
+                    float: Use a certain fraction of the features
+            min_leaf_instances (int): Minimum number of features used at each leaf
+            max_depth (int): Maximum depth to which to allow the decision trees to grow
+            randomize_pivot_location (bool): whether to draw pivots randomly or always select the midpoint
+            randomly_rotate_features (bool): whether to randomly rotate real features for each tree in the forest
+        """
+        super().__init__()
+
+        # Store the variables
+        self.num_trees = num_trees
+        self.use_jackknife = use_jackknife
+        self.subset_strategy = subset_strategy
+        self.leaf_learner = leaf_learner
+        self.min_leaf_instances = min_leaf_instances
+        self.max_depth = max_depth
+        self.randomize_pivot_location = randomize_pivot_location
+        self.randomly_rotate_features = randomly_rotate_features
+
+    def _make_learner(self):
+        if self.leaf_learner is None:
+            leaf_learner = getattr(self.gateway.jvm.io.citrine.lolo.learners.RandomForestClassifier, "$lessinit$greater$default$3")()
+        else:
+            leaf_learner = self.gateway.jvm.scala.Some(self.leaf_learner._make_learner())
+
+        return self.gateway.jvm.io.citrine.lolo.learners.RandomForestClassifier(
+            self.num_trees,
+            self.use_jackknife,
+            leaf_learner,
+            self.subset_strategy,
+            self.min_leaf_instances,
+            self.max_depth,
+            self.randomize_pivot_location,
+            self.randomly_rotate_features
+        )
+
+
+class MultiTaskRandomForest(BaseLoloRegressor):
```

Comment:
I don't love that there are now more options for the LoloPy user to choose from

Reply:
I don't think there is an alternative option. The `_make_learner` method is called before the training data is encountered, so we would not know whether to construct a single- or multi-task RF on the scala side.

And I would argue that having them split on the Scala side is appropriate. Previously, the single- and multi-task RFs were both jammed in as a `RandomForest <: Learner`, which was distinctly not a `MultiTasklearner` within the Scala inheritance chain. I think breaking them up follows a much more intuitive inheritance chain for the multi-task option.